### PR TITLE
Add Thundra Foresight Integration to the CI Build

### PR DESCRIPTION
**Pull request by rwxdash**

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -53,9 +62,19 @@ jobs:
...
+        if: ${{ github.event_name == 'push' && github.ref == 'refs/heads/master' }}
+        uses: thundra-io/thundra-gradle-test-action@v1
+        with:
+          apikey: ${{ secrets.THUNDRA_API_KEY }}
+          project_id: ${{ secrets.THUNDRA_PROJECT_ID }}
       - name: Build and test with Gradle (${{matrix.gradle_args}})
         run: |
-          ./gradlew --no-daemon --continue --scan ${{matrix.gradle_args}}
+          ./gradlew --no-daemon --continue --scan ${{matrix.gradle_args}} \
+            $($THUNDRA_GRADLE_INIT_SCRIPT_PATH || echo "--init-script $THUNDRA_GRADLE_INIT_SCRIPT_PATH")
```
Comment (aoozdemir) [inline .github/workflows/ci.yml:74]: This is the easiest way to understand if the Thundra step was successful and if so, instrument the build. If not, we don't want to break the build. Hence the check. 

The template of the init script can be found here: https://github.com/thundra-io/thundra-gradle-test-action/blob/master/templates/thundra.gradle.ejs

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -53,9 +62,19 @@ jobs:
...
+        env:
+          THUNDRA_AGENT_TEST_RUN_ID: ${{ needs.set_thundra_testrun_id.outputs.thundra_agent_testrun_id }}
+          THUNDRA_AGENT_REPORT_REST_BASEURL: https://collector.thundra.us/v1
```
Comment (aoozdemir) [inline .github/workflows/ci.yml:77]: This base URL is our staging environment. Our showroom page is fetching the data from this environment. Normally, in production, you don't need to set this. We wanted to split the showroom environment from production, that's why this is needed here.

**Review by bsideup — APPROVED** [review]

**Review by rnorth — APPROVED** [review]
Thanks @rwxdash, and sorry for not reviewing this immediately after getting back from holiday. The extra encapsulation you've done in the `thundra_test_initializer` really helps a lot - thanks!

Happy to approve, and will merge. Looking forward to seeing the output in Thundra Foresight!

**Comment by rwxdash** [comment]
Awesome! Thank you, @rnorth. No worries there, we are happy to see you on board. :)

**Pull request merged**